### PR TITLE
WIP: add API documentation

### LINKED ISSUE #1: add doxygen documentation
add all documentation from http://maemo.org/api_refs/5.0/5.0-final/icd2/index.html

### PR DESCRIPTION
**Pull request by spinal84**
Closes #1

### CODE REVIEW

**Comment by freemangordon** [comment]
hmm, this is not what the issue is for ("add doxygen documentation") - it needs to be added in source/header files as doxygen comments, see https://github.com/maemo-leste/icd2/blob/master/icd/icd_context.c for example. Not as HTML.

P.S. I'd recommend you to join #maemo-leste on freenode to coordinate the efforts, otherwise there is always a risk of misunderstanding.

**Comment by spinal84** [comment]
I see. Is that API documentation commit excessive?

**Comment by MerlijnWajer** [comment]
I think the idea is (correct me if I am wrong) that the documentation source (in doxygen format) needs to go in the source files, and then we can just use doxygen to generate the html files.

**Comment by spinal84** [comment]
Should the API docs just be temporary added to the repository until it will be integrated in the sources?  
After that the doxygen documentation will be refreshed. Are these docs somehow outdated already?

File: icd/policy_api.h [hunk 1]
```diff
@@ -45,10 +44,10 @@ struct icd_policy_request {
   /** service provider id */
   gchar *service_id;
 
-  /** (pseudo)network type */
+  /** (pseudo)network attributes */
   gchar *network_type;
 
-  /** (pseudo)network attributes */
+  /** (pseudo)network type */
```
Comment (spinal84) [inline icd/policy_api.h:66]: already fixed in local repository

File: icd/icd_script.c [hunk 2]
```diff
@@ -210,21 +208,22 @@ icd_script_exec (const gchar * script, const gchar *iface, const gchar *mode,
 }
 
 /**
- * @brief Run network script and start waiting for the result
```
Comment (spinal84) [inline icd/icd_script.c:213]: fixed

File: icd/icd_script.c [hunk 3]
```diff
@@ -103,12 +102,11 @@ icd_script_timeout_secs (void)
 }
 
 /**
- * @brief Script timeout function
+ * @brief  Script timeout function
  *
- * @param data pointer to pid of the script that timed out
```
Comment (spinal84) [inline icd/icd_script.c:108]: fixed

**Pull request closed without merging**